Makefile

```diff
@@ -22,6 +22,18 @@ $(TOOLS)/go-licenses: PACKAGE=github.com/google/go-licenses
 .PHONY: tools
 tools: $(GOLICENSES)
 
+.PHONY: test
+#: run the tests!
+test:
+ifeq (, $(shell which gotestsum))
+	@echo " ***"
+	@echo "Running with standard go test because gotestsum was not found on PATH. Consider installing gotestsum for friendlier test output!"
+	@echo " ***"
+	go test -race -v ./...
```

Comment:
This will not work across module boundaries. Please take a look at this: https://github.com/open-telemetry/opentelemetry-go/blob/86f325839ca329997672fbae8b054e7b6181b4d2/Makefile#L105-L119

Reply:
Well, shoot. OK. Will look at that and adjust. (And I'll remove the gotestsum addendum below.)

Reply:
I see the beginnings of the "[things with go.mod in them](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/105/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R5)" being added over in #105 which would make the basis for the multiple directories in which to run `go test`. I am ready to coordinate changes however y'all like.

Reply:
[The Makefile trick](https://github.com/open-telemetry/opentelemetry-go/blob/86f325839ca329997672fbae8b054e7b6181b4d2/Makefile#L105-L119)† for factorying up test/ make targets for each module seems to not do anything for the module appearing in the current (repo root) directory.

```make
ALL_GO_MOD_DIRS := $(shell find . -type f -name 'go.mod' ! -path '$(TOOLS_MOD_DIR)/*' ! -path './LICENSES/*' -exec dirname {} \; | sort)
GO_MOD_TEST_TARGETS := $(ALL_GO_MOD_DIRS:%=test/%)

test: $(GO_MOD_TEST_TARGETS)
test/%: DIR=$*
test/%:
	cd $(DIR) && go test
```

[remake](https://remake.readthedocs.io/en/latest/) shows that the target for the root `go.mod` evaluates to `test/.` (which makes sense):

```
$ remake --targets | grep test
        test
        test/.
        test/./offsets-tracker
        test/./test/e2e/gorillamux
        test/./test/e2e/nethttp
```

But make ignores that target when actually run.

```
$ make test
cd ./offsets-tracker && go test
?       go.opentelemetry.io/auto/offsets-tracker        [no test files]
cd ./test/e2e/gorillamux && go test
?       go.opentelemetry.io/auto/test/e2e/gorillamux    [no test files]
cd ./test/e2e/nethttp && go test
?       go.opentelemetry.io/auto/test/e2e/nethttp       [no test files]
```

† Which I have mad respect for personally, having been accused of Too Much Make in other projects.

Reply:
Added that ☝🏻 variation of the test targets to the Makefile in 335f20d to give us more to look at and review.